SWIPs/swip-streaming-of-feeds.md

```diff
@@ -0,0 +1,148 @@
+* [get the nearest last index of an arbitrary time](###-Nearest-Last-Index-for-an-Arbitrary-Time)
+* [construct feed topic](###-Feed-Topic-Construction)
+* [upload feed chunk](###-Upload-Feed-Chunk)
+* [download feed chunk at specific time](###-Download-Feed-Chunk-At-Specific-Time)
+* [and download (whole) feed stream](###-Download-Feed-Stream)
+
+The feeds always have a `topic` and an `index`, that we have to define.
+
+### Nearest Last Index for an Arbitrary Time
+
+Let's figure out what is the nearest last `feed index` of a given arbitrary time.
+We know the current time right now (`Tp`) which is surely greater or equal to the last upload time (`Tn`) of the feed stream.
+We also know two metadata of the stream: initial time (`T0`) and update period (`Δ1`).
+We want to find the nearest last index to an arbitrary time (`Tx`) where `T0 <= Tx <= Tn <= Tp`
+All of these are timestamps and their smallest unit is 1 second, e.g. even with this time unit it is still uncertain whether the latest update can be downloaded if `Tp = Tn`, because of the nature of P2P storage systems.
+
+So the formula is really simple how to calculate the nearest last index parameter (`i`) of the `Tx`:
+
+```ts
+function getIndexForArbitraryTime(Tx: number, T0: number, updatePeriod: number): number {
+  return Math.floor((Tx - T0) / updatePeriod) // which is the `i`
+}
+```
+
+### Feed Topic Construction
+
+The `feed topic` has to contain the initial time (`T0`) and optionally can be prefixed/suffixed with an additional identification so that the uploader with the same key can maintain many distinct feed streams.
+Either of them is chosen, the feed topic construction algorithm has to be the same and well-known between the parties. 
+
+### Upload Feed Chunk
+
+Within the upload function, the current time (`Tp`) is initialized and the `feed index` of the newest feed chunk can be calculated by calling `getIndexForArbitraryTime(Tp, T0, updatePeriod)`.
+After the `feed topic` is initialized as well in the before mentioned way, the uploading can happen as at other feed uploads. 
+
+### Download Feed Chunk at Specific Time
+
+The feed download method only has 2 required parameters, where the last one is optional: `function downloadFeed(owner: EthAddress, topic: bytes[], Tx?: number): Feed`.
+
+If the last parameter has not been passed, the `downloadFeed` function will calculate the nearest last index based on the current time like within the uploading feed function.
+
+There are cases when the chunk is not available on the first calculated index, then the lookup method starts and tries to find the nearest downloadable chunk.
+The most reasonable approach here is check the previous (`i-1`) and the next fetch index (`i+1`) until the worst case respectively `0` and last (`n`) index. 
+This lookup also can happen paralelly and checking _n_ chunks simulteniously on both sides in order to raise the certainty for the successful hit.
+
+### Download Feed Stream
```

Comment:
The proposed feed indexing method and lookup is an extension of sequential/periodic feeds, which has the advantage of not only the arbitrary topmost epoch base time, but its lookup, that pulls the closest available chunk with the _certainty_ it should be retreivable, but if not, it will surely find the quickliest retrievable chunk in this way.
This approach of the proposal also could be an extension of epoch-based feeds, but IMO that wouldn't bring any real yield, but additional constraints.